stackhpc_monasca_agent_plugins/checks/nvidia.py

```diff
@@ -0,0 +1,225 @@
+# Copyright (c) 2018 StackHPC Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import logging
+import monasca_agent.collector.checks as checks
+import pynvml
+
+
+log = logging.getLogger(__name__)
+
+_METRIC_NAME_PREFIX = "nvidia"
+_METRIC_NAME = "job_status"
+
+
+class Nvidia(checks.AgentCheck):
+    def __init__(self, name, init_config, agent_config):
+        super(Nvidia, self).__init__(name, init_config, agent_config)
+
+    def handle_not_supported(f):
```

Comment:
Is handle_not_supported decorator to accommodate for different versions of graphics cards with different API?

Reply:
That's right. If the call isn't supported it throws a 'Not Supported' exception.